blob/src/main/java/com/bazaarvoice/emodb/blob/core/DefaultBlobStore.java

```diff
@@ -408,19 +426,24 @@ public void delete(String tableName, String blobId) {
 
         StorageSummary storageSummary = _metadataProvider.readMetadata(table, blobId);
 
+        delete(table, blobId, storageSummary);
+    }
+
+    private void delete(Table table, String blobId, StorageSummary storageSummary) {
         if (storageSummary == null) {
+            LOGGER.error("Metadata isn't present for table: {}, blobId: {}", table.getName(), blobId);
             throw new BlobNotFoundException(blobId);
         }
         _metadataProvider.deleteMetadata(table, blobId);
         try {
             _storageProvider.deleteObject(table, blobId);
         } catch (Throwable t) {
-            LOGGER.error("Failed to delete blob for table: {}, blobId: {}, attempt to revert metadata deletion. Exception: {}", tableName, blobId, t.getMessage());
+            LOGGER.error("Failed to delete blob for table: {}, blobId: {}, attempt to revert metadata deletion. Exception: {}", table.getName(), blobId, t.getMessage());
             try {
                 _metadataProvider.writeMetadata(table, blobId, storageSummary);
             } catch (Exception e1) {
-                LOGGER.error("Failed to revert metadata deletion for table: {}, blobId: {}. Inconsistency between blob and metadata storages. Exception: {}", tableName, blobId, e1.getMessage());
-                _metadataInconsistencyMeter.mark();
+                LOGGER.error("Failed to revert metadata deletion for table: {}, blobId: {}. Inconsistency between blob and metadata storages. Exception: {}", table.getName(), blobId, e1.getMessage());
+                _dataNotPresentMeter.mark();
```

Comment:
Is this not a case of metadata not present, so wrong meter :?
In both put and delete cases, because of the respective order of deletions, the case seems to be only metadata missing......

Reply:
yeah, it was correct before :)